cmd/gangway/config.go

```diff
@@ -27,26 +27,32 @@ type Config struct {
 	Host string `yaml:"host"`
 	Port int    `yaml:"port"`
 
-	ClusterName   string   `yaml:"clusterName" envconfig:"cluster_name"`
-	AuthorizeURL  string   `yaml:"authorizeURL" envconfig:"authorize_url"`
-	TokenURL      string   `yaml:"tokenURL" envconfig:"token_url"`
-	ClientID      string   `yaml:"clientID" envconfig:"client_id"`
-	ClientSecret  string   `yaml:"clientSecret" envconfig:"client_secret"`
-	Audience      string   `yaml:"audience"`
-	RedirectURL   string   `yaml:"redirectURL" envconfig:"redirect_url"`
-	Scopes        []string `yaml:"scopes"`
-	UsernameClaim string   `yaml:"usernameClaim" envconfig:"username_claim"`
-	EmailClaim    string   `yaml:"emailClaim" envconfig:"email_claim"`
+	ClusterName     string   `yaml:"clusterName" envconfig:"cluster_name"`
+	AuthorizeURL    string   `yaml:"authorizeURL" envconfig:"authorize_url"`
+	TokenURL        string   `yaml:"tokenURL" envconfig:"token_url"`
+	ClientID        string   `yaml:"clientID" envconfig:"client_id"`
+	ClientSecret    string   `yaml:"clientSecret" envconfig:"client_secret"`
+	Audience        string   `yaml:"audience"`
+	RedirectURL     string   `yaml:"redirectURL" envconfig:"redirect_url"`
+	Scopes          []string `yaml:"scopes"`
+	UsernameClaim   string   `yaml:"usernameClaim" envconfig:"username_claim"`
+	EmailClaim      string   `yaml:"emailClaim" envconfig:"email_claim"`
+	UseTLS          bool     `yaml:"useTLS" envconfig:"use_tls"`
```

Comment:
Let's collapse this into a condition that checks for cert and key being present. `UseTLS` is not necessary in that case.